Helper/Config.php

```diff
@@ -163,42 +161,79 @@ public function getEnvironmentVariable(string $envVarName): ?string
      *
      * @return void
      */
-    public function setIsMerchantActive(bool $value): void
+    public function setIsMerchantActive(?int $storeId, bool $isActive): void
     {
-        $this->scopeWriter->save(self::IS_MERCHANT_ACTIVE, $value);
-        $this->flushCaches();
+        $token = $this->getAccessToken($storeId);
+        if ($token === null) { // no need to set isActive when no token exists
+            return;
+        }
+        $metadata = new ProtectMetadata($token, $isActive);
+        $this->setStoreMetadata($storeId, $metadata);
     }
 
     /**
      * Returns if the merchant is Active for NS8
      *
      * @return bool
      */
-    public function isMerchantActive(): bool
+    public function isMerchantActive(?int $storeId): bool
     {
-        return (bool) $this->scopeConfig->getValue(self::IS_MERCHANT_ACTIVE);
+        $metadata = $this->getStoreMetadata($storeId);
+        return $metadata !== null ? $metadata->isActive : false;
     }
 
     /**
-     * Retrieve an access token.
-     *
-     * @return string|null The NS8 Protect Access Token.
+     * Retrieve an access token for a given store.
+     * @param string $storeId ID of the store to retrieve token for
+     * @return string|null The NS8 Protect Access Token for the provided store.
+     */
+    public function getAccessToken(?int $storeId): ?string
+    {
+        $metadata = $this->getStoreMetadata($storeId);
+        return $metadata !== null ? $metadata->token : null;
+    }
+
+    public function setAccessToken(?int $storeId, string $token): void
+    {
+        $metadata = new ProtectMetadata($token, $this->isMerchantActive($storeId));
+        $this->setStoreMetadata($storeId, $metadata);
+    }
+
+    /**
+     * @return ProtectMetadata
+     */
+    public function getStoreMetadata(?int $storeId): ?\stdClass
+    {
+        $metadatas = $this->getStoreMetadatas();
+        $storeId = $storeId !== null ? $storeId : self::EMPTY_STORE_ID;
+        return isset($metadatas[$storeId]) ? $metadatas[$storeId] : null;
+    }
+
+    /**
+     * Get all access tokens, for all stores.
+     * Structured as { [storeId: string]: ProtectMetadata };
+     * @return ProtectMetadata[]
      */
-    public function getAccessToken(): ?string
+    public function getStoreMetadatas(): array
     {
-        return $this->encryptor->decrypt($this->scopeConfig->getValue(self::ACCESS_TOKEN_CONFIG_KEY));
+        $rawTokensJson = $this->encryptor->decrypt($this->scopeConfig->getValue(self::ACCESS_TOKEN_CONFIG_KEY));
+        $accessTokens = (array)json_decode($rawTokensJson);
```

Comment:
Instead of casting after `json_decode`, PHP may respond better to `json_decode($rawTokensJson, true)`

Reply:
this is intentional - the shallow cast makes an `Array<stdClass>` instead of `Array<Array>`, which means we can treat the items as `ProtectMetadata` instances

Reply:
Thanks for explaining! Given this, can you help me understand why we are using the `ProtectMetadata` class instead of an associative array?

Reply:
Stronger typing, pretty much. Gives us better assurance within the Config class

Reply:
Got it. i'm totally open to the new class but very hesitant about casting in this fashion. Instead of this, can the `ProtectMetadata` have added methods to support setting the values without this type of casting logic?

Reply:
I don't think so? Pretty sure that would require a lot of dependencies to be injected, and this class is just a data wrapper. What's wrong with this style of casting?

Reply:
This style of casting leaves a lot of choices up to PHP's interpreter for how the array is cast which is cause for concern. If a new developer were to change the structure of the object, I think the general assumption is that `json_decode` conversions to arrays will utilize `json_decode($x, true)` rather than shallow copies. This opens up a variety of issues if the person reusing this method expects the deep copying. Additionally, we're masquerading an stdClass object as one of the custom class, which if we expand the custom class down the line, can lead to error surrounding the method not being found or forced type checks being required.

Reply:
Instead of dependency injection, can we just map through the array and return one with PHP's intended casting?

e.g. 
```
$rawTokensJson = $this->encryptor->decrypt($this->scopeConfig->getValue(self::ACCESS_TOKEN_CONFIG_KEY))
$accessTokenValues = json_decode($rawTokensJson, true);
return array_map(function ($key, $value)  {
 return [$key] => new ProtectMetadata($value['access_token'], $value['is_active']);
}, $accessTokenValues);
```